Add dnf5 posttrans notify plugin

aleasto opened the pull request:
Just like #600, we should do the same for dnf5

aleasto commented:
cc: @gordonmessmer as you implemented the dnf4 plugin, and @ximion 

I don't know if it's acceptable to just invoke `gdbus` rather than linking some dbus library, which would probably be GIO to match the rest of packagekit.

Also I don't know if it's acceptable to add a dependency to c++ like this.

Conan-Kudo commented on meson.build:
```diff
@@ -1,8 +1,8 @@
-project('PackageKit', 'c',
+project('PackageKit', 'c', 'cpp',
```
This needs to be in the backend's `meson.build`, not here.

aleasto replied:
Thanks, done.

Conan-Kudo commented on meson.build:
```diff
@@ -1,8 +1,8 @@
   license : 'LGPL-2.1+',
   meson_version : '>=0.50',
-  default_options : ['warning_level=2', 'c_std=c99'],
+  default_options : ['warning_level=2', 'c_std=c99', 'cpp_std=c++20'],
```
This needs to be in the backend's `meson.build`, not here.

Conan-Kudo commented on backends/dnf/notify_packagekit.cpp:
```diff
@@ -0,0 +1,80 @@
+    void post_transaction(const libdnf5::base::Transaction & transaction) override;
+};
+
+void NotifyPackagekitPlugin::post_transaction(const libdnf5::base::Transaction & transaction) {
+    std::system("gdbus call --system --dest org.freedesktop.PackageKit --object-path /org/freedesktop/PackageKit --method org.freedesktop.PackageKit.StateHasChanged posttrans");
+}
```
Can you use `sdbus-cpp` here instead of shelling out to `gdbus`?

aleasto replied:
I'll see if it isn't necessarily complex.
Otherwise `gdbus` or alternatively `busctl` are pretty much guaranteed to be installed already so we wouldn't be adding new dependencies.

Conan-Kudo commented on backends/dnf/notify_packagekit.cpp:
```diff
@@ -0,0 +1,80 @@
+#include <libdnf5/base/base.hpp>
```
This needs a license header, similar to what is present in other source files.

Conan-Kudo commented on backends/dnf/notify_packagekit.cpp:
```diff
@@ -0,0 +1,80 @@
+
+namespace {
+
+constexpr const char * PLUGIN_NAME{"template"};
```
plugin name is wrong

Conan-Kudo commented on backends/dnf/notify_packagekit.cpp:
```diff
@@ -0,0 +1,80 @@
+
+constexpr const char * PLUGIN_NAME{"template"};
+
+constexpr plugin::Version PLUGIN_VERSION{.major = 1, .minor = 0, .micro = 0};
```
This should probably match PackageKit's version.

aleasto commented on backends/dnf/meson.build:
```diff
@@ -20,6 +22,26 @@ install_data(
+  cpp_args: '-std=c++20',
+  include_directories: packagekit_glib2_includes,
+  dependencies: [
+    dnf5_dep,
```
anyone familiar with meson can tell me why i don't actually need to add the dnf5 dependency to the module? it seems like headers are propagated globally? that's weird...

aleasto replied:
Ah well, because they're installed in a default includes directory

ximion commented:
@Conan-Kudo Does this look good to you now? I intend to do a new PK release tomorrow...

Conan-Kudo commented:
The Fedora dockerfile needs to be bumped to 40.

aleasto commented:
Done

sidt4 commented:
iirc, @mcrha has been working on some dnf5 stuff. Just wanted to check if there is no overlap here.

aleasto commented:
As far as I can see, @mcrha is implementing a dnf5 backend in gnome-software to ditch PackageKit altogether.

ximion commented:
> As far as I can see, @mcrha is implementing a dnf5 backend in gnome-software to ditch PackageKit altogether.

That is my understanding as well, which is slightly annoying as that moves us to a future were we yet again have to implement support for all the different PMs in many places, instead of having just one place to do the work once and for everyone.

I do understand where this is coming from though - PK could really use a lot more time investment to move it into the future.

mcrha commented:
> I do understand where this is coming from though - PK could really use a lot more time investment to move it into the future.

Correct, that's the main reason for that work.

By the way, I'm not sure this change has a meaning at the moment, when, in fact, PckageKit uses dnf4 only. Yes, one can install/update/remove with both dnf versions on a single machine, still a better option would be to truly use dnf4 or dnf5, not a mix of both.

Conan-Kudo commented:
Well, I'd like to have a dnf5 backend, since other software centers using PackageKit should benefit from it.

mcrha commented:
Sure, that makes perfect sense and Philip, the gnome-software maintainer, prefers that idea too.

aleasto commented:
> > I do understand where this is coming from though - PK could really use a lot more time investment to move it into the future.
> 
> Correct, that's the main reason for that work.
> 
> By the way, I'm not sure this change has a meaning at the moment, when, in fact, PckageKit uses dnf4 only. Yes, one can install/update/remove with both dnf versions on a single machine, still a better option would be to truly use dnf4 or dnf5, not a mix of both.

PackageKit is eventually going to be run alongside dnf5, presumably in f41 already. This change is so that dnf5 CLI operations can notify PackageKit, addressing the concerns at https://github.com/PackageKit/PackageKit/pull/578#issuecomment-1384781816
